pygfx/renderers/wgpu/_utils.py

```diff
@@ -193,6 +195,38 @@ def generate_uniform_struct(dtype_struct, structname):
     return code
 
 
+class GpuCache:
+    """A chache for GPU objects."""
+
+    _caches = {}
+
+    @classmethod
+    def get_cache_stats(cls):
+        """Get a dict mapping cache names to item counts."""
+        return {name: cache.get_count() for name, cache in GpuCache._caches.items()}
+
+    def __init__(self, name):
+        assert isinstance(name, str)
+        assert name not in GpuCache._caches
+        GpuCache._caches[name] = self
+
+        self._objects = weakref.WeakValueDictionary()
+
+    def get_count(self):
+        """Get the number of (alive) objects in the cache."""
+        return len(list(self._objects.values()))
+
+    def get(self, hash):
+        """Get the cached object or None."""
```

Comment:
I would recommend to refer to `key` instead of `hash`, since python dicts actually compute the hash of the key internally. It's a little confusing to read `hash` here, it seems to imply that you're working with a handcrafted dictionary implementation.

Reply:
👍